src/Http/Http.Features/src/FeatureReferences.cs

```diff
@@ -80,7 +80,7 @@ public void Initalize(IFeatureCollection collection, int revision)
         }
 
         // Update and cache clearing logic, when the fast-path in Fetch isn't applicable
-        private TFeature UpdateCached<TFeature, TState>(ref TFeature cached, TState state, Func<TState, TFeature> factory, int revision, bool flush) where TFeature : class?
+        private TFeature? UpdateCached<TFeature, TState>(ref TFeature? cached, TState state, Func<TState, TFeature?> factory, int revision, bool flush) where TFeature : class
```

Comment:
❔ Is it really necessary to allow `factory` to return null here?

Reply:
There's quite a few places that pass a use a factory that produces nulls.